### ISSUE TITLE
Open File and Send to Process not working on Mac OSX

### ISSUE THREAD

**Issue by arigolan** [JeffreyDavidsz/VICREO-Listener#55]
Hi, 

I'm new to Vicreo. Using v 2.0.9 on mac OSX 10.14.6 (Mojave)
Using Stream Deck and Companion 2.1.0 on a remote computer on the LAN.

I've gotten VICREO hotkey: Combination to work just fine for "Command" + N, which I've set as a Keyboard Shortcut in mimoLive to trigger "next slide". And I've got "Command" + Tab to peform it's normal function of cycling through apps (which I can use to bring mimoLive to the foreground (make it the Active app), so I'm communicating fine between the companion server (mac mini) and the remote machine (on the same LAN) running mimoLive. However, the app needs to be in the foreground for the "next slide" trigger to work. So I've tried adding the action of "Open file" for both the app itself, as well as trying it for the document (just to force it to the foreground) with no luck. I also tried  Vicreo "Send To Process" mimoLive, with no apparent results, other than an error msg in the Vicreo Log window 
"please change to new syntax"
"received ...whatever the command was..." 

I've read the now Closed thread #29 re problems on Windows eagerly thinking maybe those functions just were not working, but it seems you mention having fixed the bugs. Have you confirmed on Mac OSX that "Open File" and/or "Send to Process" are working?

I tried both with and without using " " before and after the app name, and/or the file name.
Before I read the thread, I thought maybe I was messing up the file path. I've used Terminal, Option right-clicking to copy path name, and dragging to the "Go to Folder" window- to confirm file path. I even tried adding adding the cd command that Terminal shows when doing the above over the network. All with no results.

I'm hoping this is just a bug on mac and I'm not that incompetent :)

Assuming it is, hopefully it can be easily fixed. If not, is there anything you can suggest testing, and I'll gladly report back to you?

Also, assuming thing are working as expected, is "Open File" the best method for bringing an app to the foreground? If not, please consider the feature request for a preset but to "Activate App".

Int the meantime, thanks for the awesome program, and please keep up the great work!

Ari

**Comment by JeffreyDavidsz** [comment]
First set the instance in companion to the right version of the listener you are using, that should solve some issues you are having (that's the message you received). For the process part make sure you use the right name. Not really a osx user myself but you'll probably find it in the force shutdown section of osx. Let me know if this works for you

**Comment by arigolan** [comment]
Ahh- Never noticed the version issue. That solved the issues! Thank you for the quick response.

As an FYI to any newbs that might have had an issue with file path on Mac OSX, since I was not working locally, but on a remote machine, I made the mistake of acquiring the file path remotely as well (via Terminal, and also via "option + right click copy filepath", which returned the incorrect path (it excluded "/Users/username/" (which I had "Volumes" in place of). Once I got the filepath directly (locally) from the remote machine I was trying to run it on, everything worked fine.
